docs/reference/snapshot-restore/apis/put-repo-api.asciidoc

```diff
@@ -0,0 +1,260 @@
+After restoring a source-only snapshot, you must <<docs-reindex,reindex>> the
+data into a new index.
++
+See <<snapshots-source-only-repository>>.
+
+`url`::
+URL repository. Repositories of this type are read-only
+for the cluster. This means the cluster can retrieve or restore snapshots from
+the repository but cannot write or create snapshots in it.
++
+You can use URL repositories as an alternative way to give a cluster read-only
+access to a shared file system (`fs`) repository.
++
+See <<snapshots-read-only-repository>>.
+====
+
+More repository types are available through these official
+plugins:
+
+* {plugins}/repository-s3.html[repository-s3] for S3 repository support
+* {plugins}/repository-hdfs.html[repository-hdfs] for HDFS repository support in
+  Hadoop environments
+* {plugins}/repository-azure.html[repository-azure] for Azure storage
+  repositories
+* {plugins}/repository-gcs.html[repository-gcs] for Google Cloud Storage
+  repositories
+--
+
+`settings`::
++
+--
+(Required, object)
+Contains settings for the repository. Valid properties for the `settings` object
+depend on the repository type, set using the
+<<put-snapshot-repo-api-request-type,`type`>> parameter.
+
+.Valid `settings` properties for `fs` repositories
+[%collapsible%open]
+====
+`chunk_size`::
+(Optional, <<byte-units,byte value>>)
+Maximum size of files in snapshots. In snapshots, files larger than this are
+broken down into chunks of this size or smaller. Defaults to `null` (unlimited
+file size).
+
+`compress`::
+(Optional, boolean)
+If `true`, metadata files, such as index mappings and settings, are compressed
+in snapshots. Data files are not compressed. Defaults to `true`.
+
+`location`::
+(Required, string)
+Location of the shared filesystem used to store and retrieve snapshots. This
+location must be registered in the `path.repo` setting on all master and data
+nodes in the cluster.
+
+`max_restore_bytes_per_sec`::
+(Optional, <<byte-units,byte value>>)
+Maximum snapshot restore rate per node. Defaults to `40mb` per second.
+
+`max_snapshot_bytes_per_sec`::
+(Optional, <<byte-units,byte value>>)
+Maximum snapshot creation rate per node. Defaults to `40mb` per second.
+
+`readonly`::
+(Optional, boolean)
+If `true`, the repository is read-only. The cluster can retrieve and restore
+snapshots from the repository but not write to the repository or create
+snapshots in it.
++
+If `false`, the cluster can write to the repository and create snapshots in it.
+Defaults to `false`.
++
+TIP: If you register the same snapshot repository with multiple clusters, only
+one cluster should have write access to the repository. Only a cluster with
+write access can create snapshots in the repository. All other clusters
+connected to the repository should have the `readonly` parameter set to `true`.
+This means those clusters can retrieve or restore snapshots from the repository
+but not create snapshots in it.
```

Comment:
Maybe add a sentence here, along the lines of "Having multiple clusters write to the repository at the same time risks corrupting the contents of the repository".

Reply:
Thanks! Addressed with 9cecaff01ce.